### PR TITLE
Create MaskedImageCompletionOutput and fix ViT docs

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
Fixes the output class and docs of `ViTForMaskedImageModeling `which returns images of shape (batch_size, num_channels, height, width) whereas MaskedLMOutput logits are of shape (batch_size, seq_length, vocab_size).

**Notes:**
- We have no checkpoints for `ViTForMaskedImageModeling` and the docstrings use the pretrained ViTModel combined with random head parameters. I will open a separate PR for the other affected model - `DeiTForMaskedImageModeling`, for which no checkpoints are available either. 
- Swin has its own MaskedImageOutput class but only has base model checkpoints (trained on the masked image modeling task) and no task-specific checkpoints.
- I'm planning to create a masked-image-completion pipeline and add Swin and ICT (once Sheon's PR is merged).

CC: @sheonhan

## Before submitting
- [X ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/modeling_outputs.py [hunk 1]
```diff
@@ -1281,6 +1281,34 @@ class ImageSuperResolutionOutput(ModelOutput):
...
+        attentions (`tuple(torch.FloatTensor)`, *optional*, returned when `output_attentions=True` is passed or when `config.output_attentions=True`):
+            Tuple of `torch.FloatTensor` (one for each layer) of shape `(batch_size, num_heads, patch_size,
+            sequence_length)`.
+
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
+            heads.
+    """
+
+    loss: Optional[torch.FloatTensor] = None
+    logits: torch.FloatTensor = None
```
Comment (amyeroberts) [inline src/transformers/modeling_outputs.py:1307]:
```suggestion
    reconstruction: torch.FloatTensor = None
```

**Comment by alaradirik** [comment]
Gently pinging @amyeroberts for the final approval

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for adding!

**Comment by sheonhan** [comment]
masked-image-completion pipeline sounds awesome 🙌
